feature/experimental/bgp/otg_tests/bgp_remove_private_as/bgp_remove_private_as_otg_test.go

```diff
@@ -0,0 +1,417 @@
+				_, ok := gnmi.Watch(t, otg, nbrPath.SessionState().State(), time.Minute, func(val *ygnmi.Value[otgtelemetry.E_BgpPeer_SessionState]) bool {
+					currState, ok := val.Val()
+					return ok && currState.String() == state
+				}).Await(t)
+				if !ok {
+					t.Errorf("No BGP neighbor formed for peer %s", configPeer.Name())
+				}
+			}
+		}
+	}
+}
+
+// verifyBGPAsPath is to Validate AS Path attribute using bgp rib telemetry on ATE.
+func verifyBGPAsPath(t *testing.T, otg *otg.OTG, config gosnappi.Config, asSeg []uint32, removeASPath bool) {
+	t.Helper()
+	_, ok := gnmi.WatchAll(t, otg, gnmi.OTG().BgpPeer(ateDst.Name+".BGP4.peer").UnicastIpv4PrefixAny().State(),
+		time.Minute, func(v *ygnmi.Value[*otgtelemetry.BgpPeer_UnicastIpv4Prefix]) bool {
+			return v.IsPresent()
+		}).Await(t)
+
+	if ok {
+		bgpPrefixes := gnmi.GetAll(t, otg, gnmi.OTG().BgpPeer(ateDst.Name+".BGP4.peer").UnicastIpv4PrefixAny().State())
+		gotPrefixCount := len(bgpPrefixes)
+		if gotPrefixCount != routeCount {
+			t.Errorf("Received prefixes on otg are not as expected got prefixes %v, want prefixes %v", gotPrefixCount, routeCount)
+		} else {
+			t.Logf("Received prefixes on otg are matched, got prefixes %v, want prefixes %v", gotPrefixCount, routeCount)
+		}
+	}
+
+	/*
+		// Below code will be un commented once issue 1659 is resolved for otg.
+		// https://github.com/openconfig/featureprofiles/issues/1659
+
+		var wantASSeg = []uint32{dutAS, ateAS1}
+		if removeASPath {
+			for _, as := range asSeg {
+				if as < 64512 {
+					wantASSeg = append(wantASSeg, as)
+				}
+			}
+		} else {
+			wantASSeg = append(wantASSeg, asSeg...)
+		}
+
+		var gotASSeg []uint32
+		_, ok = gnmi.WatchAll(t, otg, prefixPath.AsPathAny().State(), 2*time.Minute, func(v *ygnmi.Value[*otgtelemetry.BgpPeer_UnicastIpv4Prefix_AsPath]) bool {
+			val, present := v.Val()
+			gotASSeg = val.AsNumbers
+			return present && cmp.Diff(val.AsNumbers, wantASSeg) == ""
+		}).Await(t)
+		if !ok {
+			t.Errorf("Obtained AS path on ATE is not as expected, gotASSeg %v, wantASSeg %v", gotASSeg, wantASSeg)
+		}
+	*/
+}
+
+// TestRemovePrivateAS is to Validate that private AS numbers are stripped
+// before advertisement to the eBGP neighbor.
+func TestRemovePrivateAS(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	ate := ondatra.ATE(t, "ate")
+	otg := ate.OTG()
+	var otgConfig gosnappi.Config
+
+	t.Run("Configure DUT interfaces", func(t *testing.T) {
+		configureDUT(t, dut)
+	})
+
+	t.Run("Configure DEFAULT network instance", func(t *testing.T) {
+		dutConfNIPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut))
+		gnmi.Replace(t, dut, dutConfNIPath.Type().Config(), oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_DEFAULT_INSTANCE)
+	})
+
+	dutConfPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP")
+	t.Run("Configure BGP Neighbors", func(t *testing.T) {
+		gnmi.Delete(t, dut, dutConfPath.Config())
```

Comment:
Need to configure route policy here like in the ATE test: https://github.com/openconfig/featureprofiles/blob/e3dc053e7c0abe36b6899660c377b7f0bbe09632/feature/experimental/bgp/ate_tests/bgp_remove_private_as/bgp_remove_private_as_test.go#L339

Reply:
updated, inline with ATE test.